http-clients/url-connection-client/src/test/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClientWithCustomCreateWireMockTest.java

```diff
@@ -63,15 +68,290 @@ public void connectionsAreNotReusedOn5xxErrors() throws Exception {
 
     @Test
     public void testGetResponseCodeNpeIsWrappedAsIo() throws Exception {
-        connectionInterceptor = safeFunction(connection -> {
-            connection = spy(connection);
-            doThrow(new NullPointerException()).when(connection).getResponseCode();
-            return connection;
+        connectionInterceptor = safeFunction(connection -> new DelegateHttpURLConnection(connection) {
+            @Override
+            public int getResponseCode() {
+                throw new NullPointerException();
+            }
```

Comment:
Note to reviewer: Mockito can no longer spy on classes loaded by the bootstrap class loader in JDK 17, so we had to do the method overriding manually.